WordPress/Classes/ViewRelated/Reader/Detail/Views/ReaderDetailNewHeaderView.swift

```diff
@@ -0,0 +1,421 @@
+import SwiftUI
+import WordPressUI
+import Gridicons
+
+/// A protocol conformed by both `ReaderDetailHeaderView` and `ReaderDetailNewHeaderViewHost`.
+/// This is a temporary solution to abstract and simplify usage in the view controller.
+///
+/// NOTE: This protocol should be removed once the `readerImprovements` flag is deleted.
+///
+protocol ReaderDetailHeader: NSObjectProtocol {
+    var delegate: ReaderDetailHeaderViewDelegate? { get set }
+    var useCompatibilityMode: Bool { get set }
+
+    func configure(for post: ReaderPost)
+    func refreshFollowButton()
+}
+
+// MARK: - SwiftUI View Host
+
+class ReaderDetailNewHeaderViewHost: UIView {
+    weak var delegate: ReaderDetailHeaderViewDelegate?
+
+    // TODO: Find out if we still need this.
+    var useCompatibilityMode: Bool = false
+
+    private let isLoggedIn: Bool
+
+    // TODO: Populate this with values from the ReaderPost.
+    private lazy var viewModel: ReaderDetailHeaderViewModel = {
+        $0.topicDelegate = self
+        $0.headerDelegate = delegate
+        return $0
+    }(ReaderDetailHeaderViewModel())
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    init(isLoggedIn: Bool) {
+        self.isLoggedIn = isLoggedIn
+
+        super.init(frame: .zero)
+        setupView()
+    }
+
+    func setupView() {
+        translatesAutoresizingMaskIntoConstraints = false
+
+        let headerView = ReaderDetailNewHeaderView(viewModel: viewModel) { [weak self] in
+            guard let swiftUIView = self?.subviews.first else {
```

Comment:
This causes a warning since the variable is unused.
```suggestion
            guard self?.subviews.first != nil else {
```

Reply:
Thanks! This is fixed in https://github.com/wordpress-mobile/WordPress-iOS/pull/21674/commits/4623b4da0d404e893acebb61bf4bbfb360cc42db.